models/container.go

```diff
@@ -2,49 +2,71 @@ package models
 
 import (
 	"bytes"
+	"io"
 
 	"github.com/astaxie/beego"
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/stdcopy"
 	"golang.org/x/net/context"
 )
 
-var (
-	Containers map[string]*Container
-)
+type Logs struct {
+	ContainerLog       containerLog       `json:"containerLog"`
+	Base64ContainerLog base64ContainerLog `json:"base64ContainerLog"`
+}
 
-type Container struct {
-	containerID string
-	Logs        string
+type containerLog struct {
+	ContainerID string `json:"containerID"`
+	Logs        string `json:"Logs"`
 }
 
-func init() {
-	Containers = make(map[string]*Container)
+type base64ContainerLog struct {
+	Base64containerID string `json:"base64containerID"`
+	Base64Logs        []byte `json:"base64Logs"`
 }
 
-func GetLog(containerID string, tail string) (string, error) {
+func GetLog(containerID string, tail string) (Logs, error) {
```

Comment:
TIP: for larger projects, you should decouple your `models` package (where your types are defined) from your business logic (how the models are used). The reason being, that there will be a lot of packages in your project that need to interact with your data models, but they don't all need to know about how to fetch, persist or process them.